pkg/operator/target_config_reconciler.go

```diff
@@ -262,6 +273,11 @@ func checkProfileConflicts(profiles sets.String, profileName deschedulerv1.Desch
 	if profiles.Has(string(deschedulerv1.DevPreviewLongLifecycle)) && profiles.Has(string(deschedulerv1.LifecycleAndUtilization)) {
 		return fmt.Errorf("cannot declare %s and %s profiles simultaneously, ignoring", deschedulerv1.DevPreviewLongLifecycle, deschedulerv1.LifecycleAndUtilization)
 	}
+
+	if scheduler.Spec.Profile == configv1.HighNodeUtilization &&
+		(profiles.Has(string(deschedulerv1.LifecycleAndUtilization)) || profiles.Has(string(deschedulerv1.DevPreviewLongLifecycle))) {
+		return fmt.Errorf("enabling Descheduler LowNodeUtilization with Scheduler HighNodeUtilization may cause an eviction/scheduling hot loop")
```

Comment:
This can get tricky as it's only one directional validation. E.g. The scheduler profile is set to LowNodeUtilization and the descheduler profile set to LifecycleAndUtilization. Then, a user changes the scheduler profile to HighNodeUtilization at which point the descheduler operator will start reporting error and changes its state to Degraded eventually. Which is fine as long as the user notices that before the descheduler is triggered. What about, instead of reporting the error, disable the descheduler operand and change the operator state to Degraded right away?

Reply:
@ingvagabund good points, I made some changes in a new commit to compare... need to test it but what I did was:
* Move the conflict check out of the loop it was in (this wasn't related, but it only needs to be checked once)
* If that failed, return an error from `manageConfigMap` but *also* return `forceDeployment == true`
* With `err != nil && forceDeployment`, update the deployment to be 0 replicas
* Update operator status with new deployment generation and degraded=true condition

wdyt?

Reply:
I like it!!!